src/classes/FeatureParameterMapper_TEST.cls

```diff
@@ -66,18 +66,150 @@ private class FeatureParameterMapper_TEST {
         System.assertEquals(true, featureParamMapperInstance1 != null, 'Instance of service class should not be null.');
     }
 
+    /**************************************************************************************************************************
+    * @description Test method to verify getFeatureParameterType returns a map
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterType() {
+
+        Test.startTest();
+             Map<FeatureParameterMapper.TelemetryParameterName, String> dataTypeByFeature = FeatureParameterMapper.getInstance().getFeatureParameterType();
+        Test.stopTest();
+
+        System.assertNotEquals(null, dataTypeByFeature.size(), 'The map with the values should be retrieved.');
+    }
+
     /**************************************************************************************************************************
     * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
     ***************************************************************************************************************************/
     @isTest
-    private static void getFeatureParameterModel() {
+    private static void getFeatureParameterModelStringNotAnEnum() {
+
+        try {
+            Test.startTest();
+                FeatureParameterModel featureParamModel = FeatureParameterMapper.getInstance().getFeatureParameterModel('testFeatureParam');
+                System.assert(false, 'When Passed parameter string is not an enum an error is thrown');
+            Test.stopTest();
+        } catch (FeatureParameterMapper.FeatureParameterMapperException ex) {
+            System.assert(ex.getMessage().contains('Could not find the Feature Parameter testFeatureParam'));
+        }
+
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterModelEnumNotDate() {
+        try {
+            Test.startTest();
+                FeatureParameterModel featureParamModel = FeatureParameterMapper.getInstance().getFeatureParameterModel('HasUserManagedTDTM');
+                System.assert(false, 'When Passed parameter string is not an enum an error is thrown');
+            Test.stopTest();
+        } catch (FeatureParameterMapper.FeatureParameterMapperException ex) {
+            System.assert(ex.getMessage().contains('Feature Parameter'));
+            System.assert(ex.getMessage().contains('is not defined.'));
```

Comment:
@MatthewBlanski would this be an ideal way to handle asserts? Do we want to rearrange the exception in main class to accommodate the entire string in 1 assert?